tests/ci/integration/sslproxy_patch/aws-lc-sslproxy.patch

```diff
@@ -0,0 +1,115 @@
+diff --git a/tests/check/cachedsess.t.c b/tests/check/cachedsess.t.c
+index 7dbac6e..58a47d5 100644
+--- a/tests/check/cachedsess.t.c
++++ b/tests/check/cachedsess.t.c
+@@ -40,6 +40,8 @@
+ 
+ #if defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x20501000L
+ #define TMP_SESS_FILE "pki/session-libressl-2.5.0.pem"
++#elif defined(AWSLC_API_VERSION) && AWSLC_API_VERSION == 22
```

Comment:
Why the specific check for API version 22? Did we add support in version 23? If so should this be <= 22?

Reply:
`22` is the current version we're on. This test tests against the dumped out contents of `SSL_SESSION` and checks if it's reparsable. This isn't standardized behavior however, and we don't make any promises that contents `SSL_SESSION` won't change. The test understands this and only compares against specific versions of libcrypto.
* https://github.com/sonertari/SSLproxy/blob/30ed5b420002c8f1ac6f5c865373baa1e46bc4b8/tests/check/cachedsess.t.c#L41

I'm setting the specific `AWSLC_API_VERSION` here to help the test detect incompatibility, whenever we update the contents of our `SSL_SESSION`. There isn't a guarantee that older versions of AWS-LC are compatible with the current `session-aws-lc-1.15.0.pem` file, which is why I used `== 22`.

Reply:
Do you have a mechanism to remember this? I'm concerned we'll bump our API version for some other reason which will turn off this test. Then in the future we could forget to update this and/or the test.

Also if we do remember to update this test is it documented how to generate this test file?

Reply:
Added `sslproxy_patch_reminder` in the bash script to remind us, along with steps to regenerate the test file.